Feat/audio recorder audio form

louvrecly opened the pull request:
### Descriptions
- Add form behaviour  to `VoiceForm`
- Ref: [`react-hook-form`](https://react-hook-form.com/)

### Stories
- [x] Remove `play` method from `AudioData` in `audioRecorder`
- [x] Make `audio` param in `setAudio` method optional
- [x] Add `embeddedRecordButton` to `IComponentState`
- [x] Add `embedRecordButton` to `IComponentAction`
- [x] Refactor `RecordButton` to show normal or embedded styles according to `embeddedRecordButton` state
- [x] Refactor `AudioPlayer` to receive `audioUrl` prop from parent and render `RecordButton` when `audio` is removed
- [x] Install `react-hook-form`
- [x] Define `VoiceFormData` interface
- [x] Register `VoiceForm` with `react-hook-form` to call `submitVoice` on `submit`
- [x] Add `onTouchStart` and `onTouchEnd` event handlers to capture touch events for non-desktop devices
- [x] Add `disableContextMenu` to `onContextMenu` to prevent the context menu from long press events in non-desktop devices
- [x] Add `/ab-test-web` prefix to all url paths

thrinithomas commented:
I have encounter a bug (problem?) related to github-pages between react-router-dom,
whilst the application redirect user to `/`, it will redirect to `https://louvrecly.github.io/` rather than `https://louvrecly.github.io/ab-test-web`. 

Therefore I suggest we put `ab-test-web/` prefix in every url path.

louvrecly commented:
> I have encounter a bug (problem?) related to github-pages between react-router-dom,
> whilst the application redirect user to `/`, it will redirect to `https://louvrecly.github.io/` rather than `https://louvrecly.github.io/ab-test-web`.
> 
> Therefore I suggest we put `ab-test-web/` prefix in every url path.

Added [here](https://github.com/louvrecly/ab-test-web/pull/32/files#diff-e97dc9d2a0f65ec8914e0d105b261a98R40-R46) as suggested! 

I have deployed the updates to [github pages](https://louvrecly.github.io/ab-test-web) and added a new env variable `REACT_APP_URL_PREFIX` to [secrets](https://github.com/louvrecly/ab-test-web/settings/secrets/new)!
This PR is ready for review again! 👍
